libkbfs/disk_block_cache.go

```diff
@@ -45,17 +57,29 @@ type DiskBlockCacheStandard struct {
 
 var _ DiskBlockCache = (*DiskBlockCacheStandard)(nil)
 
-func newDiskBlockCacheStandard(config diskBlockCacheConfig, dirPath string,
-	maxBytes uint64) (*DiskBlockCacheStandard, error) {
+// openLevelDB opens or recovers a leveldb.DB with a passed-in storage.Storage
+// as its underlying storage layer.
+func openLevelDB(stor storage.Storage) (db *leveldb.DB, err error) {
+	db, err = leveldb.Open(stor, leveldbOptions)
+	if _, isErrCorrupted := err.(*storage.ErrCorrupted); isErrCorrupted {
+		return leveldb.Recover(stor, leveldbOptions)
```

Comment:
Yeah, definitely worth logging. I'll do that. This can happen if a leveldb isn't closed, while the manifest is being written to. It's harmless other than that the recover takes much longer than a regular open (not sure how long, I'd have to benchmark with some database size).